src/btree/bt_random.c

```diff
@@ -69,8 +70,14 @@ __random_skip_entries(WT_CURSOR_BTREE *cbt, WT_INSERT_HEAD *ins_head)
 
     /* Find a level with enough entries on it to predict the size of the list. */
     for (level = WT_SKIP_MAXDEPTH - 1; level >= 0; --level) {
-        for (entries = 0, t = &ins_head->head[level]; *t != NULL; t = &(*t)->next[level])
+        entries = 0;
+        t = &ins_head->head[level];
+        WT_C_MEMMODEL_ATOMIC_LOAD(temp, t, WT_ATOMIC_RELAXED);
```

Comment:
Here I have to extract the statements from the for loop because the macros cannot be used in the for statement.

The macros have to be defined in this way because I have to implement the macros for [windows](https://github.com/wiredtiger/wiredtiger/pull/9121/files#diff-8522ce6c3f3398c56477194bdff525a273c3284054719711580b4c4e40c0aef2R45).

If we don't need to worry about windows, we can have something like:

temp = WT_C_MEMMODEL_ATOMIC_LOAD(t, WT_ATOMIC_RELAXED);

This would allow us to use this also in the for loop.